### ISSUE TITLE
Julia 1.8.4, 1.8.5 and 1.9.0-beta3 crash with XGBoost.jl and LIBSVM.jl under Windows 11

### ISSUE THREAD

**Issue by mlesnoff** [JuliaLang/julia#48187]
I am using LIBSVM.jl v0.8.0, and XGBoost.jl v2.2.0, under Windows 11.

It works fine with Julia 1.8.3:

```
julia> versioninfo()
Julia Version 1.8.3
Commit 0434deb161 (2022-11-14 20:14 UTC)
Platform Info:
OS: Windows (x86_64-w64-mingw32)
CPU: 16 × Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
WORD_SIZE: 64
LIBM: libopenlibm
LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
Threads: 8 on 16 virtual cores
Environment:
JULIA_EDITOR = code
JULIA_NUM_THREADS = 8
```

But when I uses Julia 1.8.4:

```
julia> versioninfo()
Julia Version 1.8.4
Commit 00177ebc4f (2022-12-23 21:32 UTC)
Platform Info:
OS: Windows (x86_64-w64-mingw32)
CPU: 16 × Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
WORD_SIZE: 64
LIBM: libopenlibm
LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
Threads: 8 on 16 virtual cores
Environment:
JULIA_EDITOR = code
JULIA_NUM_THREADS = 8
```
running any function of LIBSVM.jl or XGBoost.jl kills my Julia session, for instance after doing:
```
using XGBoost
(X, y) = (randn(100,4), randn(100))
```
then the command below;
`bst = xgboost((X, y))`

kills the process and closes directly Julia without any printed error in the terminal.

I created a thread at [Discourse](https://discourse.julialang.org/t/issue-with-xgboost-jl-and-libsvm-jl-when-julia-1-8-4/92396/2) but got no solution. 

I already opened two issues: [https://github.com/JuliaML/LIBSVM.jl/issues/95](https://github.com/JuliaML/LIBSVM.jl/issues/95) and [https://github.com/dmlc/XGBoost.jl/issues/153](https://github.com/dmlc/XGBoost.jl/issues/153). The second raised unexplained error on windows.

Both  LIBSVM.jl and XGBoost.jl  wrap C/C++ code; this may be linked.

Did somebody observe the same problem and know what could happen?

**Comment by mlesnoff** [comment]
The problem seems to persist with Julia 1.8.5 and 1.9 betas [](https://discourse.julialang.org/t/issue-with-xgboost-jl-and-libsvm-jl-when-julia-1-8-4/92396/7)

**Comment by mlesnoff** [comment]
We still observe the same issue on v1.9.0-beta3: under Windows11 Julia crashes when using any function of XGBoost.jl or LIBSVM.jl (both are C/C++ wrappers)

[Discourse](https://discourse.julialang.org/t/issue-with-xgboost-jl-and-libsvm-jl-when-julia-1-8-4/92396/10)

**Comment by stefanjwojcik** [comment]
I'm having the same issue on Windows 11 and Julia 1.8.5, trying to fit XGBoost.

**Comment by laborg** [comment]
I'm relying on LIBSVM on Windows but this regression is quite a show stopper. For me it occurs on Windows 10.

**Comment by giordano** [comment]
CC @staticfloat

**Comment by jeremiedb** [comment]
FWIW - XGBoost.jl is still crashing on Julia v1.9.0-rc1 on Windows

**Comment by ablaom** [comment]
This seems related: https://discourse.julialang.org/t/issue-with-xgboost-jl-and-libsvm-jl-when-julia-1-8-4/92396/40?u=jd-foster

**Comment by brenhinkeller** [comment]
Apparently still happening in 1.9.1: https://julialang.slack.com/archives/C67TK21LJ/p1686195803822999

**Comment by giordano** [comment]
Should be fixed by #50135
